### ISSUE TITLE
fsrtlregisterfilesystemfiltercallbacks

### ISSUE THREAD

**Issue by rodwiddowson** [MicrosoftDocs/windows-driver-docs-ddi#937]
This [page](https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/ntifs/nf-ntifs-fsrtlregisterfilesystemfiltercallbacks )
This [github location](https://github.com/MicrosoftDocs/windows-driver-docs-ddi/blob/staging/wdk-ddi-src/content/ntifs/nf-ntifs-fsrtlregisterfilesystemfiltercallbacks.md)

@lorihollasch , @DOMARS 

I discovered to my chagrin that the behavior of this API when faced with bad input (a zero length) changed in ` 10.0.18363.1016` (with a crash replacing "working" behavior whena  failure status might have been better).

I think that this is a software bug, but it probably needs documented if it isn't fixable.  I'll pull together  PR if you think that it's worthwhile.

**Comment by lorihollasch** [comment]
Hi @rodwiddowson , can you be more specific about the bad zero-length input? FsRtlRegisterFileSystemFilterCallbacks is expecting 2 input pointers. Are you passing zero for Callbacks.SizeOfFsFilterCallbacks?

**Comment by rodwiddowson** [comment]
@lorihollasch I'm away from my sources but if that is the the size of the callback structure that's it,

I had ExAllocatePoolZero'd the table and only set up the stat entry point (but not the size).  The registration work but the first time that we tried to dispatch through the table we get a bugcheck,  I could probably dig out a stack if that would help.

Thanks!

**Comment by lorihollasch** [comment]
@rodwiddowson , sorry for delayed response! I've updated the page to say that SizeOfFsFilterCallbacks cannot be zero. (I actually created new separate pages for the structures associated with FsRtlRegisterFileSystemFilterCallbacks since that page was so long and hard to parse.) If you are still able, feel free to send a stack trace to Craig with the error that you hit (so that he doesn't have to look for a needle in a haystack :).
